optaplanner-benchmark/src/test/java/org/optaplanner/benchmark/impl/DefaultPlannerBenchmarkTest.java

```diff
@@ -0,0 +1,95 @@
+        ExecutorService executorService = mock(ExecutorService.class);
+        BenchmarkReport benchmarkReport = mock(BenchmarkReport.class);
+
+        PlannerBenchmarkResult benchmarkResultWithEmptySolverResultList = new PlannerBenchmarkResult();
+
+        DefaultPlannerBenchmark benchmark = new DefaultPlannerBenchmark(benchmarkResultWithEmptySolverResultList,
+                                                                        solverConfigContext, benchmarkDirectory,
+                                                                        executorService, executorService, benchmarkReport);
+
+        assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(benchmark::benchmarkingStarted)
+                .withMessageStartingWith("The solverBenchmarkResultList").withMessageEndingWith("cannot be empty.");
+    }
+
+    @Test
+    public void throwIllegalArgumentException_WhenBenchmarkDirectoryIsNull() {
+        SolverConfigContext solverConfigContext = mock(SolverConfigContext.class);
+        ExecutorService executorService = mock(ExecutorService.class);
+        BenchmarkReport benchmarkReport = mock(BenchmarkReport.class);
+        SolverBenchmarkResult benchMarkResult = mock(SolverBenchmarkResult.class);
+
+        PlannerBenchmarkResult plannerBenchmarkResult = new PlannerBenchmarkResult();
+        plannerBenchmarkResult.setSolverBenchmarkResultList(Collections.singletonList(benchMarkResult));
+
+        DefaultPlannerBenchmark benchmark = new DefaultPlannerBenchmark(plannerBenchmarkResult,
+                                                                        solverConfigContext, null,
+                                                                        executorService, executorService, benchmarkReport);
+        assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(benchmark::benchmarkingStarted)
+                .withMessageStartingWith("The benchmarkDirectory").withMessageEndingWith("must not be null.");
+    }
+
+    @Test
+    public void propagateExceptionMessage_WhenExceptionThrownDuringWarmUp() {
+        SolverConfig solverConfig = PlannerTestUtils.buildSolverConfig(
+                TestdataSolution.class, TestdataEntity.class);
+        PlannerBenchmarkFactory benchmarkFactory = PlannerBenchmarkFactory.create(
+                PlannerBenchmarkConfig.createFromSolverConfig(solverConfig));
+
+        TestdataSolution solution = mock(TestdataSolution.class);
+        String exceptionMessage = "Message to be received with ExecutionException call.";
+        when(solution.getEntityList()).thenThrow(new UnsupportedOperationException(exceptionMessage));
+
+        DefaultPlannerBenchmark benchmark = (DefaultPlannerBenchmark) benchmarkFactory.buildPlannerBenchmark(solution);
+
+        try {
+            benchmark.benchmark();
+        } catch (PlannerBenchmarkException e) {
+            assertEquals(exceptionMessage, e.getCause().getMessage());
```

Comment:
instead of checking the exception message, you can check if e.getCause() is the same instance as you originally have thrown

Reply:
As we decided to keep AssertJ and it's already used to checking exception in the method above this one, let's keep it consistent and use the same approach here.